pkg/awsutils/awsutils.go

```diff
@@ -410,25 +403,24 @@ func (cache *EC2InstanceMetadataCache) refreshSGIDs(mac string) error {
 		return errors.Wrap(err, "get instance metadata: failed to retrieve security-group-ids")
 	}
 
-	sgIDs  := strings.Fields(metadataSGIDs)
+	sgIDs := strings.Fields(metadataSGIDs)
 
 	newSGs := StringSet{}
 	newSGs.Set(sgIDs)
-	addedSGs   := newSGs.Difference(&cache.securityGroups)
+	addedSGs := newSGs.Difference(&cache.securityGroups)
 	deletedSGs := cache.securityGroups.Difference(&newSGs)
 
 	if !addedSGs.IsEmpty() {
```

Comment:
Nit: I think we should remove this `IsEmpty()` check, and the similar one a few lines below.  I think the inner `for` loop will already do the right thing (nothing) if the list is empty, and fewer branches == simpler code.